subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c

```diff
@@ -968,8 +969,46 @@ int json_send_to_cloud(cJSON *const request)
 
 	return err;
 }
+
+static int shadow_connection_reset(cJSON *connection_obj)
+{
+	int ret;
+
+	ret = json_add_null_cs(connection_obj, NRF_CLOUD_JSON_KEY_KEEPALIVE);
+	ret += json_add_null_cs(connection_obj, NRF_CLOUD_JSON_KEY_PROTOCOL);
+	ret += json_add_null_cs(connection_obj, NRF_CLOUD_JSON_KEY_METHOD);
+
+	return ret;
+}
+
 #endif /* CONFIG_NRF_CLOUD_MQTT */
 
+static int shadow_connection_update(cJSON *connection_obj)
+{
+	int ret;
+
+#if defined(CONFIG_NRF_CLOUD_MQTT)
+	ret = json_add_num_cs(connection_obj, NRF_CLOUD_JSON_KEY_KEEPALIVE,
+			      CONFIG_NRF_CLOUD_MQTT_KEEPALIVE);
+#else
+	ret = json_add_null_cs(connection_obj, NRF_CLOUD_JSON_KEY_KEEPALIVE);
+#endif
+	ret += json_add_str_cs(connection_obj, NRF_CLOUD_JSON_KEY_PROTOCOL,
```

Comment:
something like this (and one for method):
```
#if defined(CONFIG_NRF_CLOUD_MQTT)
  #define NRF_CLOUD_JSON_VAL_CFGD_PROTO_VAL   NRF_CLOUD_JSON_VAL_PROTO_MQTT
#elif defined(CONFIG_NRF_CLOUD_REST)
  #define NRF_CLOUD_JSON_VAL_CFGD_PROTO_VAL   NRF_CLOUD_JSON_VAL_PROTO_REST
#elif defined(CONFIG_NRF_CLOUD_COAP)
  #define NRF_CLOUD_JSON_VAL_CFGD_PROTO_VAL   NRF_CLOUD_JSON_VAL_PROTO_COAP
#else
  #define NRF_CLOUD_JSON_VAL_CFGD_PROTO_VAL   "Unknown"
#endif
```
would make this section easier on the eyes